### DISCUSSION TITLE
Should we migrate to Typer?

### DISCUSSION THREAD

**Discussion by Kludex** [encode/httpx#2369] [Ideas] [answered]
That's the question.

Pros would be a cute interface, better user experience, easy to read code (more maintainable).
Cons would be another dependency, but we already use an `extra` for the CLI.

Ref.: https://typer.tiangolo.com/

I can implement the transition if wanted.

**Comment by lovelydinosaur** [comment, marked as answer]
> cute interface, better user experience

We could consider specific UI improvements...?

> Cons would be another dependency

Honest answer is that I wouldn't be wild about adding the extra abstraction and dependency.

Might be interesting to take a look at how the reworking would look, but I don't think I'd want to commit to it upfront.
